PortFxCop/src/Microsoft.ApiDesignGuidelines.Analyzers/Core/EnumsShouldHaveZeroValue.Fixer.cs

```diff
@@ -1,36 +1,166 @@
-// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Generic;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CodeFixes;
-using Microsoft.CodeAnalysis.Editing;
 using System.Collections.Immutable;
-using System.Composition;
+using System.Diagnostics;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
 
 namespace Microsoft.ApiDesignGuidelines.Analyzers
-{                              
+{
     /// <summary>
     /// CA1008: Enums should have zero value
     /// </summary>
     public abstract class EnumsShouldHaveZeroValueFixer : CodeFixProvider
     {
-        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(EnumsShouldHaveZeroValueAnalyzer.RuleId);
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(EnumsShouldHaveZeroValueAnalyzer.RuleId); }
+        }
 
-        public sealed override FixAllProvider GetFixAllProvider()
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
```

Comment:
This method is new but has code from what was originally in a GetUpdateDocumentAsync method before. The big difference from a perf standpoint is that this method returns the codefixes with a delegate to perform the action, whereas the GetUpdatedDocumentAsync method before had no facility to say don't the entire codefix to know if there's a codefix available.